I am going to use the Consumption_of_alcohol_in_Russia_2017-2023.csv dataset acquired through Kaggle. (URL to dataset: https://www.kaggle.com/datasets/scibearia/consumption-of-alcohol-in-russia-2017-2023?resource=download)

In [20]:
# import required libraries in order to use queries to answer questions.
import pandas as pd
import matplotlib.pyplot as plt

data = pd.read_csv("Consumption of alcoholic bevs in RUS 2017-2023.csv")
# print sum of NULL data per column
print(data.isnull().sum())

Region                                                              0
Year                                                                0
Wine                                                                0
Beer                                                                0
Vodka                                                               0
Sparkling wine                                                      0
Brandy                                                              0
Сider                                                               0
Liqueurs                                                            0
Total alcohol consumption (in liters of pure alcohol per capita)    0
dtype: int64


Question 1: Total alcohol consumption in liters of pure alcohol by year per capita

In [2]:
yearly_consumption = data.groupby('Year')['Total alcohol consumption (in liters of pure alcohol per capita)'].mean().reset_index()

print(yearly_consumption)

   Year  Total alcohol consumption (in liters of pure alcohol per capita)
0  2017                                           5.803529               
1  2018                                           6.200000               
2  2019                                           6.168235               
3  2020                                           6.390588               
4  2021                                           6.481176               
5  2022                                           6.602353               
6  2023                                           6.790588               


Question 2: Region that consumed the most Wine and the year.

In [5]:
max_wine_consumption = data['Wine'].max()

max_wine = data[data['Wine'] == max_wine_consumption]

print(max_wine[['Region', 'Year', 'Wine']])

                     Region  Year  Wine
20  Nenets Autonomous Okrug  2017  8.47


As the region and year that consumed the most wine in the dataset, we have got The Nenets Autonomous Okrug region in 2017.

Question 3: Region that consumed the most Beer and the year.

In [6]:
max_beer_consumption = data['Beer'].max()

max_beer = data[data['Beer'] == max_beer_consumption]

print(max_beer[['Region', 'Year', 'Beer']])

             Region  Year    Beer
163  Primorsky Krai  2018  116.63


As the region and year that consumed the most beer in the dataset, we have got The Primorsky Krai region in 2018.

Question 4: Region that consumed the most Vodka and the year.

In [7]:
max_vodka_consumption = data['Vodka'].max()

max_vodka = data[data['Vodka'] == max_vodka_consumption]

print(max_vodka[['Region', 'Year', 'Vodka']])

              Region  Year  Vodka
337  Sakhalin Oblast  2020  12.83


As the region and year that consumed the most vodka in the dataset, we have got The Sakhalin Oblast region in 2020.

Question 5: Region that consumed the most Sparkling Wine and the year.

In [8]:
max_spark_wine_consumption = data['Sparkling wine'].max()

max_spark_wine = data[data['Sparkling wine'] == max_spark_wine_consumption]

print(max_spark_wine[['Region', 'Year', 'Sparkling wine']])

             Region  Year  Sparkling wine
82  Sakhalin Oblast  2017            3.36


As the region and year that consumed the most sparkling wine in the dataset, we have got The Sakhalin Oblast region in 2017.

Question 6: Region that consumed the most Brandy and the year.

In [28]:
max_brandy_consumption = data['Brandy'].max()

max_brandy = data[data['Brandy'] == max_brandy_consumption]

print(max_brandy[['Region', 'Year', 'Brandy']])

              Region  Year  Brandy
194  Murmansk Oblast  2019    2.08


As the region and year that consumed the most brandy in the dataset, we have got The Murmansk Oblast region in 2019.

Question 7: Region that consumed the most Cider and the year.

In [17]:
max_cider_consumption = data['Сider'].max()

max_cider = data[data['Сider'] == max_cider_consumption]

print(max_cider[['Region', 'Year', 'Сider']])

            Region  Year  Сider
6  Kostroma Oblast  2017   2.52


As the region and year that consumed the most cider in the dataset, we have got The Kostroma Oblast region in 2017.

Question 8: Region that consumed the most Liqueurs and the year.

In [18]:
max_liqueurs_consumption = data['Liqueurs'].max()

max_liqueurs = data[data['Liqueurs'] == max_liqueurs_consumption]

print(max_liqueurs[['Region', 'Year', 'Liqueurs']])

                  Region  Year  Liqueurs
528  Republic of Karelia  2023      2.95


As the region and year that consumed the most liqueurs in the dataset, we have got The Republic of Karelia region in 2023.

Question 9: Region that consumed the most alcohol in pure alcohol and the year

In [19]:
max_alcohol_consumption = data['Total alcohol consumption (in liters of pure alcohol per capita)'].max()

max_alcohol = data[data['Total alcohol consumption (in liters of pure alcohol per capita)'] == max_alcohol_consumption]

print(max_alcohol[['Region', 'Year', 'Total alcohol consumption (in liters of pure alcohol per capita)']])

                  Region  Year  \
273  Republic of Karelia  2020   

     Total alcohol consumption (in liters of pure alcohol per capita)  
273                                               12.4                 


As the region and year that consumed the most liqueurs in the dataset, we have got The Republic of Karelia region in 2020.

Question 10: Statistic of region that consumed the most alcohol in pure alcohol

In [27]:
test = data.groupby('Region')['Total alcohol consumption (in liters of pure alcohol per capita)'].max()
max = test.max()
max_data = data[data['Total alcohol consumption (in liters of pure alcohol per capita)'] == max]
print(max_data)

                  Region  Year  Wine   Beer  Vodka  Sparkling wine  Brandy  \
273  Republic of Karelia  2020  7.23  89.78  11.13            1.39    1.52   

     Сider  Liqueurs  \
273   0.92      1.49   

     Total alcohol consumption (in liters of pure alcohol per capita)  
273                                               12.4                 


The statistic for the region that consumed the most alcohol in pure alcohol is:


Region: Republic of Karelia

Year: 2020

Wine: 7.23

Beer: 89.78

Vodka: 11.13

Sparkling Wine: 1.39

Brandy: 1.52

Cider: 0.92

Liqueurs: 1.49

Total Alcohol Consumption: 12.4